### DISCUSSION TITLE
画像が表示されないバグの調査

### DISCUSSION THREAD

**Discussion by kagomen** [lgtm-factory/lgtm-factory#170] [Scrap]
#169 の調査です

**Comment by kagomen** [comment]
> src
以下のどちらかを指定する必要があります：
> - [静的にインポート](https://ja.next-community-docs.dev/docs/app-router/building-your-application/optimizing/images#%E3%83%AD%E3%83%BC%E3%82%AB%E3%83%AB%E3%81%AE%E7%94%BB%E5%83%8F)された画像ファイル
> - パス文字列。これは、[loader](https://ja.next-community-docs.dev/docs/app-router/api-reference/components/image#loader) prop の値により、外部の絶対 URL か内部パスのどちらかになります
外部 URL を使用する場合は、next.config.jsの[remotePatterns](https://ja.next-community-docs.dev/docs/app-router/api-reference/components/image#remotepatterns)に追加する必要があります。

https://ja.next-community-docs.dev/docs/app-router/api-reference/components/image

**Comment by kazzyfrog** [comment]
仮説として、
1. そもそも、`<Image>`コンポーネントは、vercel以外（今回ならcloudflare）で使用できるか？ →Yes
2. `<Image>`コンポーネントは、vercel以外（今回ならcloudflare）で最適化が行われているのか？ →No？（Web上で画像のサイズを確認したところ、リポジトリ内のサイズと同じだった！なので、別に最適化されていない）
3. `<Image>`コンポーネントの最適化機能は、vercelのみで使える →Yes？[ref](https://vercel.com/docs/image-optimization#optimized-url-format)
    - これが、要はローカルサーバーと、本番はvercelのみで使える仕様。vercel以外では、デプロイで詰まる
4. なので、 `<Image>`コンポーネントによる最適化はできないので、[設定でオプトアウト](https://nextjs.org/docs/app/api-reference/components/image#unoptimized)する必要があるかも。これが悪さしている気がする。。

**Reply by kazzyfrog** [reply]
もし最適化したい場合は、自前で実装するか、Cloudflareの画像最適化の方を使うかなど、、選ぶ感じかな

**Reply by kagomen** [reply]
なるほど……！
調査ありがとうございます🙇🙇
これっぽいですねーー

**Reply by kagomen** [reply]
@kazzyfrog 

next.configに
```
images: {
    unoptimized: true,
  },
```
を追加したら、`npm run preview`でも画像反映されましたー！！！！🥳

**Reply by kazzyfrog** [reply]
おお！！よかったです！！！👍

**Comment by kagomen** [comment]
> もし最適化したい場合は、自前で実装するか、Cloudflareの画像最適化の方を使うか

サクッと調べてみた感じだと、無料でCloudflareの画像最適化サービスを使うのは難しそう……？
https://cloudflare.domore.co.jp/knowledge/cloudflare-images/
<img width="800" alt="スクリーンショット 2024-09-01 16 53 09" src="https://github.com/user-attachments/assets/81b47b98-f51a-4ff3-8267-0abdc96d7746">

こんな感じで自前で実装してる方もいました
https://zenn.dev/sora_kumo/articles/wasm-image-optimization
